lua/git-worktree/init.lua

```diff
@@ -6,43 +6,149 @@ local Status = require("git-worktree.status")
+            git_worktree_root = nil
+            return
+        else
+            if stdout == "true" then
+                is_in_worktree = true
+            end
+        end
+    end
+
+    local find_git_dir_job = Job:new({
+        'git', 'rev-parse', '--git-dir',
+        cwd = cwd,
+    })
+
+    local process_find_git_dir = function(stdout, code)
+        if is_in_worktree then
+            -- if in worktree git dir returns absolute path
+
+            -- try to find the dot git folder (non-bare repo)
+            local git_dir = Path:new(stdout)
+            local has_dot_git = false
+            for _, dir in ipairs(git_dir:_split()) do
+                if dir == ".git" then
+                    has_dot_git = true
+                    break
+                end
+            end
+
+            if has_dot_git then
+                if stdout == ".git" then
+                    git_worktree_root = cwd
+                else
+                    local start = stdout:find("%.git")
+                    git_worktree_root = stdout:sub(1,start - 2)
+                end
+            else
+                local start = stdout:find("/worktrees/")
+                git_worktree_root = stdout:sub(0, start - 1)
+            end
+        elseif stdout == "." then
+            -- we are in the root git dir
+            git_worktree_root = cwd
+        else
+            -- if not in worktree git dir returns relative path
+            local start = stdout:find(".git")
+            git_worktree_root = Path:new(
+            string.format("%s" .. Path.path.sep .. "%s", cwd, stdout:sub(1,start))
+            )
+        end
+    end
+
+    local find_toplevel_job = Job:new({
+        'git', 'rev-parse', '--show-toplevel',
+        cwd = cwd,
+    })
+
+    local process_find_toplevel = function(stdout, code)
+        current_worktree_path = stdout
+    end
+
+
+    local stdout, code = inside_worktree_job:sync()
+    if code ~= 0 then
+        -- not in a git dir
+        git_worktree_root = nil
+        current_worktree_path = nil
+        return
+    end
+    stdout = table.concat(stdout, "")
+    process_inside_worktree(stdout, code)
+
+    stdout, code = find_git_dir_job:sync()
+    stdout = table.concat(stdout, "")
+    process_find_git_dir(stdout, code)
+
+    stdout, code = find_toplevel_job:sync()
+    if code ~= 0 then
+        current_worktree_path = nil
+        return
+    end
+    stdout = table.concat(stdout, "")
+    process_find_toplevel(stdout, code)
+
+end
+
+M._find_git_root_job()
+
+local function on_tree_change_handler(op, metadata)
     if M._config.update_on_change then
         if op == Enum.Operations.Switch then
-            local changed = M.update_current_buffer()
+            local changed = M.update_current_buffer(metadata["prev_path"])
             if not changed then
-                vim.cmd(string.format(":Ex %s", M.get_worktree_path(path)))
+                vim.cmd(string.format("e %s", M.get_worktree_path(metadata["path"])))
```

Comment:
I just noticed this PR which partially addresses my proposal from #33. Using the edit command would resolve the dependence on netrw and allow using any file explorer plugin.
Nonetheless, a configuration option for the command would be very welcome. See my proposed patch in the issue linked above.
Note also, that `e .` would be sufficient in the example above, which again would allow not requiring `path`.